### PR TITLE
ci: version to 4.0.0 Micronaut 4.0.0-SNAPSHOT githubCoreBranch=4.0.x

### LINKED ISSUE #414: Update to latest Kotlin version e.g. 1.7.10
### Issue description

https://kotlinlang.org/docs/releases.html#release-details

### PR DESCRIPTION
**Pull request by wetted**
closes #395 
closes #399 
closes #398 
closes #414 

satisfies task in https://github.com/micronaut-projects/micronaut-core/issues/7720

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
 
-projectVersion=3.3.0-SNAPSHOT
+projectVersion=4.0.0-SNAPSHOT
 projectGroup=io.micronaut.kotlin
 
 micronautDocsVersion=2.0.0
-micronautVersion=3.3.4
-micronautTestVersion=3.1.1
+micronautVersion=4.0.0-SNAPSHOT
+micronautTestVersion=3.4.0
```
Comment (sdelamo) [inline gradle.properties:8]: this should be 3.5.0

**Review by sdelamo — CHANGES_REQUESTED** [review]
Why does it fail?

**Comment by wetted** [comment]
> Why does it fail?

@sdelamo This has several issues I need to address for the kotlin/ktor upgrades. It's a WIP.

**Comment by wetted** [comment]
#403 and #413 to upgrade ktor are a bit more involved with breaking changes to dependency coordinates, package names and API. I will work this on a separate PR.

**Review by sdelamo — APPROVED** [review]

**Pull request merged**